lib/compat/wordpress-6.5/class-gutenberg-rest-hooked-blocks-controller-6-5.php

```diff
@@ -0,0 +1,307 @@
+			}
+			$hooked_block_types[ $anchor_block_type ] = $hooked_block_types_for_anchor_block;
+		}
+
+		// Retrieve the list of registered collection query parameters.
+		$registered = $this->get_collection_params();
+		$namespace  = '';
+		if ( isset( $registered['namespace'] ) && ! empty( $request['namespace'] ) ) {
+			$namespace = $request['namespace'];
+		}
+
+		foreach ( $hooked_block_types as $anchor_block_type => $hooked_block_types_for_anchor_block ) {
+			if ( $namespace ) {
+				list ( $block_namespace ) = explode( '/', $anchor_block_type );
+
+				if ( $namespace !== $block_namespace ) {
+					continue;
+				}
+			}
+			$data[ $anchor_block_type ] = $this->prepare_response_for_collection( $hooked_block_types_for_anchor_block );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if a given request has access to read a hooked block type.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		return true;
+
+		$check = $this->check_read_permission();
+		if ( is_wp_error( $check ) ) {
+			return $check;
+		}
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether a given hooked block type should be visible.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return true|WP_Error True if the block type is visible, WP_Error otherwise.
+	 */
+	protected function check_read_permission() {
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
+
+		return new WP_Error( 'rest_block_type_cannot_view', __( 'Sorry, you are not allowed to manage block types.' ), array( 'status' => rest_authorization_required_code() ) );
+	}
+
+	/**
+	 * Get the block, if the name is valid.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $name Block name.
+	 * @return WP_Block_Type|WP_Error Block type object if name is valid, WP_Error otherwise.
+	 */
+	protected function get_block( $name ) {
+		$block_type = $this->block_registry->get_registered( $name );
+		if ( empty( $block_type ) ) {
+			return new WP_Error( 'rest_block_type_invalid', __( 'Invalid block type.' ), array( 'status' => 404 ) );
+		}
+
+		return $block_type;
+	}
+
+	/**
+	 * Get the correct context for the request.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $entity Context entity.
+	 * @param string $id     Context ID.
+	 * @return WP_Block_Template|WP_Post|WP_Errpr Block type object if name is valid, WP_Error otherwise.
+	 */
+	protected function get_context( $entity, $id ) {
+		$entity_mapping = array(
```

Comment:
Not sure we need this logic. These values are essentially post types and we can get them from the URL where we get the $id as well in the site editor.